### PR TITLE
scrape: Add previous error to the API

### PR DESCRIPTION
**Pull request by roidelapluie**
First step towards #7384

Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

File: scrape/target.go [hunk 1]
```diff
@@ -264,6 +274,30 @@ func (t *Target) LastScrapeDuration() time.Duration {
 	return t.lastScrapeDuration
 }
 
+// PreviousError returns the error encountered during the last errorful scrape.
```
Comment (brian-brazil) [inline scrape/target.go:277]: The mix of last and previous here is confusing.

Comment (roidelapluie) [reply scrape/target.go:277]: Yes but the v1 HTTP API (used by dashboards and UIs) is considered stable, so I have to keep last in the JSON API.

Maybe I can change to `current` or `lastCompleted` in the go variables?

Comment (brian-brazil) [reply scrape/target.go:277]: The Go should follow the JSON, or that'd also be confusing.

**Comment by roidelapluie** [comment]
WDYT about " Saved Error " ?

**Comment by roidelapluie** [comment]
I have adapted the wordings

**Comment by brian-brazil** [comment]
Most recent maybe?

**Comment by roidelapluie** [comment]
target.MostRecentFailedScrapeDuration is a long var name but we can change that in the future - aka prometheus 3.0

File: docs/querying/api.md [hunk 2]
```diff
@@ -428,9 +428,9 @@ $ curl http://localhost:9090/api/v1/targets
         "lastError": "",
         "lastScrape": "2017-01-17T15:07:44.723715405+01:00",
         "lastScrapeDuration": 0.050688943,
-        "savedError": "data does not end with # EOF",
-        "savedFailedScrape": "2017-01-17T15:02:44.723715405+01:00",
-        "savedFailedScrapeDuration": 0.010683421,
+        "mostRecentError": "data does not end with # EOF",
```
Comment (brian-brazil) [inline docs/querying/api.md:431]: mostRecentFailedScrapeError would be more consistent

**Pull request closed without merging**